rust/worker/src/execution/operators/flush_s3.rs

```diff
@@ -57,6 +57,24 @@ impl Operator<FlushS3Input, FlushS3Output> for FlushS3Operator {
     }
 
     async fn run(&self, input: &FlushS3Input) -> Result<FlushS3Output, Self::Error> {
+        // TODO: Ideally we shouldn't even have to make an explicit call to
+        // write_to_blockfiles since it is not the workflow for other segments
+        // and is exclusive to metadata segment. We should figure out a way
+        // to make this call a part of commit itself. It's not obvious directly
+        // how to do that since commit is per partition but write_to_blockfiles
+        // only need to be called once across all partitions combined.
+        let mut writer = input.metadata_segment_writer.clone();
```

Comment:
nit: naming of writer may be confusing, 

also we clone this again lower down? can we for cleanliness just do it once (haven't checked the mutability needs)